onnxruntime/test/onnx/onnx_model_info.cc

```diff
@@ -70,3 +85,46 @@ OnnxModelInfo::OnnxModelInfo(_In_ const PATH_CHAR_TYPE* model_url) : model_url_(
   }
   RepeatedPtrFieldToVector(graph.output(), output_value_info_);
 }
+
+void OnnxModelInfo::InitOrtModelInfo(_In_ const PATH_CHAR_TYPE* model_url) {
+  std::vector<uint8_t> bytes;
+  size_t num_bytes = 0;
+  const auto model_location = ToWideString(model_url);
+  ORT_THROW_IF_ERROR(Env::Default().GetFileLength(model_location.c_str(), num_bytes));
+  bytes.resize(num_bytes);
+  std::ifstream bytes_stream(model_location, std::ifstream::in | std::ifstream::binary);
+  bytes_stream.read(reinterpret_cast<char*>(bytes.data()), num_bytes);
```

Comment:
I think this change will also double our unitests running time. Because every model need be read twice.

Reply:
Oh, this is only for .ort file format, which is very rare right now, the .onnx model code path is not changed.
Right now most of the flatbuffers serialize/deserialize code are in graph module, I have a plan to move most ort flatbuffers code to a separate module, such that you do not have to reconstruct the graph to get input/output information

---
In reply to: [486037539](https://github.com/microsoft/onnxruntime/pull/5100#discussion_r486037539) [](ancestors = 486037539)

Reply:
Other than that, I highly suggest we do never use fstream in our code, especially for reading anything larger than 1MB. It's a long story, we can talk offline.